Source/CarthageKit/Project.swift

```diff
@@ -537,9 +537,9 @@ public final class Project {
 		let resolver = Resolver(
 			versionsForDependency: versionsForProject,
 			cartfileForDependency: cartfileForDependency,
-			resolvedGitReference: { _, refName in
-				// Dependencies should be fully resolved already.
-				return SignalProducer(value: PinnedVersion(refName))
+			resolvedGitReference: { project, reference in
+				return resolveReferenceInRepository(repositoryFileURLForProject(project), reference)
```

Comment:
These should be fully resolved already. Is this because we're wanting to get a SHA instead of a tag name? Could a comment here and below explaining why it's necessary?